padinfo/view/pantheon.py

```diff
@@ -56,37 +62,106 @@ async def deserialize(cls, dgcog, user_config: UserConfig, ims: dict):
         reaction_list = ims.get('reaction_list')
 
         return cls(original_author_id, menu_type, raw_query, query, user_config.color, monster, alt_monsters,
-                   pantheon_list, series_name,
+                   pantheon_list, series_name, base_monster,
                    use_evo_scroll=use_evo_scroll,
                    reaction_list=reaction_list,
                    extra_state=ims)
 
+    @classmethod
+    def make_series_name(cls, series_name, filter_strings):
+        return '{} [Filters: {}]'.format(series_name, ' '.join(filter_strings))
+
     @classmethod
     async def query(cls, dgcog, monster):
+        # filtering rules:
+        # 1. don't show monsters that only have mat types (or show only mats if monster is a mat)
+        # 2. if still too many, show only monsters with the same base rarity
+        # 3. if still too many, show only monsters with the same base main attribute (any monster
+        #        with nil main attribute is compared using its subattribute)
+        # 4. if still too many, show only monsters that match both base attributes
+        # 5. if still too many, truncate (and probably redefine pantheon)
+
         db_context = dgcog.database
+        series_name = monster.series.name_en
         full_pantheon = db_context.get_monsters_by_series(monster.series_id)
         if not full_pantheon:
-            return None, None
-        pantheon_list = list(filter(lambda x: db_context.graph.monster_is_base(x), full_pantheon))
-        if len(pantheon_list) == 0 or len(pantheon_list) > 20:
-            return None, None
-
-        series_name = monster.series.name_en
-
-        return pantheon_list, series_name
+            return None, None, None
+
+        base_mon = db_context.graph.get_base_monster(monster)
+
+        base_list = [m for m in full_pantheon if db_context.graph.monster_is_base(m)]
+        if len(base_list) > 0 and len(base_list) <= MAX_MONS_TO_SHOW:
```

Comment:
These checks can be written like one of the following:
```py
if 0 < len(base_list) <= MAX_MONS_TO_SHOW:
if len(base_list) in range(1, MAX_MONS_TO_SHOW+1):
```

Reply:
the `range` check is definitely less clear, don't do that.

meh on the first one, I don't really care either way.